tools/system_libs.py

```diff
@@ -33,11 +33,14 @@ def calculate(temp_files, in_temp, stdout_, stderr_, forced=[]):
   # Check if we need to include some libraries that we compile. (We implement libc ourselves in js, but
   # compile a malloc implementation and stdlibc++.)
 
-  def read_symbols(path, exclude=None):
-    symbols = map(lambda line: line.strip().split(' ')[1], open(path).readlines())
-    if exclude:
-      symbols = filter(lambda symbol: symbol not in exclude, symbols)
-    return set(symbols)
+  def read_symbols(path, exclude=()):
+    symbols = set()
+    with open(path, 'r') as file:
+      for line in file:
+        mark, symbol = line.strip().split(' ')
+        if mark != 'U' and symbol not in exclude:
```

Comment:
This isn't needed. You can just pass the right args to ``llvm-nm`` when generating the symbols files.  Without verifying it, I think the flags to pass were ``-defined-only -extern-only``.